File: ruby/tests/base_to_base_test.rb
# frozen_string_literal: true

require 'test_helper'
require 'base_to_base'

describe BaseToBase do
  def setup
    @tests = [
      { value: 'E', base: 16, new_base: 2, want: '1110' },
      { value: '11011110101011011011111011101111', base: 2, new_base: 3, want: '100122100210211112102' },
      { value: '3735928559', base: 10, new_base: 4, want: '3132223123323233' },
      { value: '8831A383B', base: 12, new_base: 16, want: 'DEADBEEF' }
    ]
  end

  describe 'base_to_base' do
    it 'converts any base to other base' do
      @tests.each do |tc|
        _(BaseToBase.base_to_base(tc[:value], tc[:base], tc[:new_base])).must_equal(tc[:want])
      end
    end
  end
end

File: elixir/test/polyglot/base_to_base_test.exs
defmodule Polyglot.BaseToBaseTest do
  use ExUnit.Case
  alias Polyglot.BaseToBase

  setup do
    [
      tests: [
        {"E", 16, 2, "1110"},
        {"11011110101011011011111011101111", 2, 3, "100122100210211112102"},
        {"3735928559", 10, 4, "3132223123323233"},
        {"8831A383B", 12, 16, "DEADBEEF"}
      ]
    ]
  end

  describe "base_to_base" do
    test "converts any base other base", %{tests: tests} do
      for {value, base, new_base, want} <- tests do
        assert BaseToBase.base_to_base(value, base, new_base) == want
      end
    end
  end
end
